frontend/packages/dev-console/src/components/import/pipeline/PipelineSection.tsx

```diff
@@ -15,7 +17,21 @@ type PipelineSectionProps = {
 const PipelineSection: React.FC<PipelineSectionProps> = ({ flags }) => {
   const { values } = useFormikContext<FormikValues>();
 
-  if (flags[FLAG_OPENSHIFT_PIPELINE]) {
+  const canListPiplines = useAccessReview({
+    group: PipelineModel.apiGroup,
+    resource: PipelineModel.plural,
+    namespace: CLUSTER_PIPELINE_NS,
+    verb: 'list',
+  });
+
+  const canCreatePiplines = useAccessReview({
+    group: PipelineModel.apiGroup,
+    resource: PipelineModel.plural,
+    namespace: getActiveNamespace(),
+    verb: 'create',
+  });
```

Comment:
I think we also want to see if we can create `PipelineResources`, we [create them at the same time](https://github.com/openshift/console/blob/master/frontend/packages/dev-console/src/components/import//pipeline/pipeline-template-utils.ts#L45-L46). I doubt there _can_ be different permissions involved, but I think we should look to validate all the calls that are used so we don't have an issue in the future (or if you can control permissions that granular).

Recommend you refactor your checks into a "check for Pipeline creation permissions" method. That way it's isolated and this if-statement below doesn't become 40 boolean checks :)

Reply:
```useAccessReview``` is a hook and we can not call it inside a method. we have to check it separately. I added a check for pipeline resources creation as well.

Reply:
I thought Hooks could be called in any number of methods, it just had to always be invoked in the same order. Just start the method name with `use` to allow the linter to help out.

Reply:
@andrewballantyne I added the method to handle all the checks.